drivers/net/ethernet/intel/idpf/idpf_eth_idc.c

```diff
@@ -0,0 +1,453 @@
+
+/**
+ * idpf_eth_idc_event_send - Called by an Auxiliary Driver to send event
+ * to Main Driver
+ * @dev_info: Ethernet device information struct
+ * @event: Event information
+ *
+ * Ethernet driver sends the event to main driver.
+ */
+static void idpf_eth_idc_event_send(struct idpf_eth_idc_dev_info *dev_info,
+				    struct idpf_eth_idc_event *event)
+{
+	struct idpf_adapter *adapter;
+
+	adapter = container_of(dev_info->eth_shared, struct idpf_adapter,
+			       eth_shared);
+	idpf_recv_eth_event(adapter, event);
+}
+
+/**
+ * idpf_eth_idc_virtchnl_send - Called by an Auxiliary Driver
+ * @dev_info: Device information
+ * @params: Virtchnl parameters
+ */
+static
+size_t idpf_eth_idc_virtchnl_send(struct idpf_eth_idc_dev_info *dev_info,
+				  struct idpf_vc_xn_params *params)
+{
+	struct idpf_adapter *adapter;
+
+	adapter = container_of(dev_info->eth_shared, struct idpf_adapter,
+			       eth_shared);
+	return idpf_vc_xn_exec(adapter, params);
+}
+
+/**
+ * idpf_eth_idc_intr_reg_init - Called by an Auxiliary Driver
+ * @dev_info: Device information
+ * @num_vecs: Number of vectors
+ * @q_vectors: Queue vectors info
+ * @q_vector_idxs: Queue vectors index info
+ */
+static int idpf_eth_idc_intr_reg_init(struct idpf_eth_idc_dev_info *dev_info,
+				      u16 num_vecs,
+				      struct idpf_q_vector *q_vectors,
+				      u16 *q_vector_idxs)
+{
+	struct idpf_adapter *adapter;
+
+	adapter = container_of(dev_info->eth_shared, struct idpf_adapter,
+			       eth_shared);
+	return adapter->dev_ops.reg_ops.intr_reg_init(adapter, num_vecs,
+						      q_vectors, q_vector_idxs);
+}
+
+/**
+ * idpf_eth_idc_intr_init_vec_idx - Called by an Auxiliary Driver
+ * @dev_info: Device information
+ * @num_vecs: Number of vectors
+ * @q_vectors: Queue vectors info
+ * @q_vector_idxs: Queue vectors index info
+ */
+static
+int idpf_eth_idc_intr_init_vec_idx(struct idpf_eth_idc_dev_info *dev_info,
+				   u16 num_vecs,
+				   struct idpf_q_vector *q_vectors,
+				   u16 *q_vector_idxs)
+{
+	struct idpf_adapter *adapter;
+
+	adapter = container_of(dev_info->eth_shared, struct idpf_adapter,
+			       eth_shared);
+	return idpf_intr_init_vec_idx(adapter, num_vecs,
+				      q_vectors, q_vector_idxs);
+}
+
+/**
+ * idpf_eth_idc_req_rel_vector_indexes - Called by an Auxiliary Driver
+ * @dev_info: Device information
+ * @num_vectors: Number of vectors
+ * @vec_info: Vecor information
+ * @msix_table: MSIX table
+ */
+static
+int idpf_eth_idc_req_rel_vector_indexes(struct idpf_eth_idc_dev_info *dev_info,
+					u16 *num_vectors,
+					struct idpf_vector_info *vec_info,
+					struct msix_entry *msix_table)
+{
+	struct idpf_adapter *adapter;
+
+	adapter = container_of(dev_info->eth_shared, struct idpf_adapter,
+			       eth_shared);
+	return idpf_req_rel_vector_indexes(adapter, num_vectors, vec_info,
+					   msix_table);
+}
+
+extern struct idpf_eth_idc_auxiliary_driver *idpf_eth_get_driver(void);
+/**
+ * idpf_eth_idc_get_driver - returns eth driver ops
```

Comment:
please check all the comment headers. Many places there is a mismatch...

Reply:
Let me know if I missed something. Needs review.